src/main/java/com/launchacademy/reviews/services/ReviewFormService.java

```diff
@@ -0,0 +1,62 @@
+package com.launchacademy.reviews.services;
+
+import com.launchacademy.reviews.exceptions.TrailNotFoundException;
+import com.launchacademy.reviews.models.Review;
+import com.launchacademy.reviews.models.ReviewForm;
+import com.launchacademy.reviews.models.ReviewImage;
+import com.launchacademy.reviews.models.Trail;
+import com.launchacademy.reviews.models.User;
+import java.util.Optional;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+public class ReviewFormService {
+  private final TrailService trailService;
+  private final UserService userService;
+  private final ReviewService reviewService;
+  private final ReviewImageService reviewImageService;
+
+  @Autowired
+  public ReviewFormService(TrailService trailService,
+      UserService userService, ReviewService reviewService,
+      ReviewImageService reviewImageService) {
+    this.trailService = trailService;
+    this.userService = userService;
+    this.reviewService = reviewService;
+    this.reviewImageService = reviewImageService;
+  }
+
+  public Review processForm(ReviewForm reviewForm) {
+    Integer trailId = reviewForm.getTrailId();
+    Trail trail = trailService.findById(trailId)
+        .orElseThrow(() -> new TrailNotFoundException(trailId));
+
+    String userName = reviewForm.getName();
+    Optional<User> userResult = userService.findByName(userName);
+    User user;
+    if (userResult.isEmpty()) {
+      user = new User();
+      user.setName(userName);
+      userService.save(user);
+    }
+    else {
+      user = userResult.get();
+    }
+
```

Comment:
Nice work putting this in the service! Keeps the controller clean and all business logic centralized.